src/System.CommandLine/Parsing/StringExtensions.cs

```diff
@@ -101,9 +101,15 @@ internal static (string? Prefix, string Alias) SplitPrefix(this string rawAlias)
                             ? arg.Substring(1, colonIndex - 1) // [name:value]
                             : arg.Substring(1, arg.Length - 2); // [name] is a legal directive
 
-                        CliDirective? directive = knownTokens.TryGetValue(directiveName, out var directiveToken)
-                            ? (CliDirective)directiveToken.Symbol!
-                            : null;
+                        CliDirective? directive;
+                        if (knownTokens.TryGetValue($"[{directiveName}]", out var directiveToken))
```

Comment:
So the actual parse error fix was to wrap the directive name in `[]`? 

May I ask why have you decided to move `CliDirective` from config to root command then? I am not against it, I am just curious whether overall you got to a conclusion that it's where it belongs or whether I can't see some other aspect of the actual fix.

Reply:
This change isn't ideal but it's a step toward reducing the impact of a larger bug (#2223). It can't be fully fixed without incorporating directives into the symbol model, which is why moving them to `CliRootCommand` at this point was helpful. It's a design decision we discussed a few times for the sake of consistency. Tactically, this was a good time to do it since it related to this bug.

Reply:
>  It can't be fully fixed without incorporating directives into the symbol model

Thanks, now I know all I needed! 👍